COMIN/COMOUT updates for enkf chgres and downstream product jobs

KateFriedman-NOAA opened the pull request:
# Description

This PR makes COMIN/COMOUT variable updates needed for issue #2451. The changes cover the chgres_forenkf and most remaining downstream product jobs. Follow-up PRs will cover COM variable changes for the other remaining unconverted jobs, as documented here https://docs.google.com/spreadsheets/d/1S4deVHNEgSR_6-q2lDOaXq4kW-4SmxurxH-q5iPVd6o/edit?pli=1&gid=0#gid=0.

~~Setting PR as draft while testing runs on WCOSS2.~~

Refs #2451

# Type of change

Maintenance

# Change characteristics

JJOB and ex-script updates to COM variables

# How has this been tested?

Ran WCOSS2 CI suite

KateFriedman-NOAA committed:
Update COM variables for gdas_atmos_chgres_forenkf

Refs #2451

KateFriedman-NOAA committed:
Update COM variables in gfs_atmos_postsnd

Refs #2451

KateFriedman-NOAA committed:
Update COM variables in post manager

Refs #2451

KateFriedman-NOAA committed:
Update COM variables in gempak jobs

Refs #2451

KateFriedman-NOAA committed:
Update COM variables for AWIPS and downstream products

Refs #2451

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/comin_comout

RussTreadon-NOAA reviewed (dismissed):
The changes to `jobs/JGDAS_ATMOS_CHGRES_FORENKF` and `scripts/exgdas_atmos_chgres_forenkf.sh` are straight forward.  `COM` is replaced with `COMIN` or `COMOUT` as appropriate.

Approve the chgres-forenkf changes pending confirmation from g-w CI that the changes work as intended.

KateFriedman-NOAA committed:
Change COMOUT to COMIN for COM_ATMOS_HISTORY_MEM

Refs #2451

KateFriedman-NOAA commented:
> The changes to `jobs/JGDAS_ATMOS_CHGRES_FORENKF` and `scripts/exgdas_atmos_chgres_forenkf.sh` are straight forward. `COM` is replaced with `COMIN` or `COMOUT` as appropriate.

Thanks @RussTreadon-NOAA ! Upon further review I changed a COMOUT in those scripts to a COMIN. Committed @ c5647916c. Please re-review that job update. Thanks!

RussTreadon-NOAA reviewed (dismissed):
Approve pending confirmation from g-w CI that the changes work as intended.

CoryMartin-NOAA reviewed (dismissed):
I agree with Russ, EnKF scripts look good to me, approve pending testing

WenMeng-NOAA reviewed (dismissed):
The updates in the ATMOS_POST and ATMOS_PRODUCTS parts look good to me. Approve pending testing.

WenMeng-NOAA commented:
@KateFriedman-NOAA Could you invite @ChristopherHill-NOAA to review the updates related to the GFS ATMOS downstream jobs?

KateFriedman-NOAA commented:
> @KateFriedman-NOAA Could you invite @ChristopherHill-NOAA to review the updates related to the GFS ATMOS downstream jobs?

@WenMeng-NOAA I added @ChristopherHill-NOAA , thanks! Should we set Chris as the code owner for the downstream job scripts now? See this file: https://github.com/NOAA-EMC/global-workflow/blob/develop/.github/CODEOWNERS

WenMeng-NOAA commented:
> > @KateFriedman-NOAA Could you invite @ChristopherHill-NOAA to review the updates related to the GFS ATMOS downstream jobs?
> 
> @WenMeng-NOAA I added @ChristopherHill-NOAA , thanks! Should we set Chris as the code owner for the downstream job scripts now? See this file: https://github.com/NOAA-EMC/global-workflow/blob/develop/.github/CODEOWNERS

@KateFriedman-NOAA Yes, please switch the code owner from @GwenChen-NOAA to @ChristopherHill-NOAA. Thanks!

BoCui-NOAA commented:
@KateFriedman-NOAA 

Please makes variable updates in ush/gfs_bufr.sh:

change COM_ATMOS_HISTORY to COMIN_ATMOS_HISTORY
change COM_ATMOS_BUFR to COMOUT_ATMOS_BUFR

Thanks!

KateFriedman-NOAA committed:
Update COM variables in gfs_bufr.sh

Refs #2451

KateFriedman-NOAA commented:
> @KateFriedman-NOAA
> 
> Please makes variable updates in ush/gfs_bufr.sh:
> 
> change COM_ATMOS_HISTORY to COMIN_ATMOS_HISTORY change COM_ATMOS_BUFR to COMOUT_ATMOS_BUFR
> 
> Thanks!

Thanks for catching my oversight @BoCui-NOAA ! I have made the updates @ fd0d6292d.

KateFriedman-NOAA committed:
Update CODEOWNERS file for downstream products

- Change codeowner for downstream product scripts
from Gwen Chen to Christopher Hill

Refs #2451

BoCui-NOAA commented:
> > @KateFriedman-NOAA
> > Please makes variable updates in ush/gfs_bufr.sh:
> > change COM_ATMOS_HISTORY to COMIN_ATMOS_HISTORY change COM_ATMOS_BUFR to COMOUT_ATMOS_BUFR
> > Thanks!
> 
> Thanks for catching my oversight @BoCui-NOAA ! I have made the updates @ [fd0d629](https://github.com/NOAA-EMC/global-workflow/commit/fd0d6292d1ade755d9be5ed9e1aceee827e44bb6).

Got the updates. Thanks!

KateFriedman-NOAA commented:
> Yes, please switch the code owner from @GwenChen-NOAA to @ChristopherHill-NOAA. Thanks!

@WenMeng-NOAA thanks, I have updated the CODEOWNERS file @ c67c23926.

BoCui-NOAA commented on jobs/JGFS_ATMOS_POSTSND:
```diff
@@ -21,17 +21,20 @@ export DBNROOT=${DBNROOT:-${UTILROOT}/fakedbn}
-
-if [[ ! -d "${COM_ATMOS_BUFR}" ]]; then
-    mkdir -p "${COM_ATMOS_BUFR}"
+YMD=${PDY} HH=${cyc} declare_from_tmpl -rx \
+  COMIN_ATMOS_HISTORY:COM_ATMOS_HISTORY_TMPL \
+  COMOUT_ATMOS_BUFR:COM_ATMOS_BUFR_TMPL \
+  COMOUT_ATMOS_WMO:COM_ATMOS_WMO_TMPL \
+  COMOUT_ATMOS_GEMPAK:COM_ATMOS_GEMPAK_TMPL
```
What are these directories COM_ATMOS*_TMPL for? The names are not consistent, for example, COMIN_ATMOS_HISTORY vs COM_ATMOS_HISTORY_TMPL? @KateFriedman-NOAA

aerorahul replied:
@BoCui-NOAA 
`declare_from_tmpl` creates the variable `COMIN_ATMOS_HISTORY` based on the variable template defined under `COM_ATMOS_HISTORY_TMPL`.  `COM_ATMOS_HISTORY_TMPL` is declared in `parm/config/gfs/config.com`.  
The result of this call in effect is:
```bash
export COMIN_ATMOS_HISTORY=$(compath.py ${envir}/com/gfs/${gfs_ver})"/${RUN}.${PDY}/${cyc}/model/atmos/history"
```

BoCui-NOAA replied:
Got it now, Thanks!

KateFriedman-NOAA committed:
Update CODEOWNERS file

Refs #2451

ChristopherHill-NOAA commented on .github/CODEOWNERS:
@WenMeng-NOAA @KateFriedman-NOAA   In the midst of updating`.github/CODEOWNERS`, I recommend adding @AminIlia-NOAA as a co-owner for the following files:
```
jobs/JGLOBAL_OCEANICE_PRODUCTS @ChristopherHill-NOAA
scripts/exglobal_oceanice_products.py @ChristopherHill-NOAA
scripts/exglobal_oceanice_products.py @ChristopherHill-NOAA
oceanice_nc2grib2.sh @ChristopherHill-NOAA
ush/python/pygfs/task/oceanice_products.py @aerorahul @ChristopherHill-NOAA

```

GwenChen-NOAA replied:
I am the original developer of the ocean/ice product scripts. I would like to retain the ownership of the ocean/ice product scripts as a co-owner. Thanks!

KateFriedman-NOAA replied:
@GwenChen-NOAA @AminIlia-NOAA @ChristopherHill-NOAA please see the updates I just committed @ e2370c624 and let me know if further `CODEOWNERS` updates are needed. Thanks!

GwenChen-NOAA replied:
Thanks, @KateFriedman-NOAA!  Looks good to me.

@ChristopherHill-NOAA, do you also work on ocean/ice products? Or just downstream awips and gempak products?

ChristopherHill-NOAA replied:
@GwenChen-NOAA  I helped to identify runtime errors with multiple products from HR4, but I will continue to focus primarily on maintaining the code for atmospheric products.

GwenChen-NOAA replied:
Understood. Thanks!

ChristopherHill-NOAA reviewed (dismissed):
Each of the `COM_ATMOS_*` directory declarations are now distinguished between `COMIN_ATMOS_*` and `COMOUT_ATMOS_*` directories according to the reference of input or output data, respectively.  Most definitions of the COMIN and COMOUT directories remain controlled by the corresponding `COM_ATMOS_*_TMPL` designation that continues to exist within configuration file `config.com`.

Provided nominal results to CI testing, the proposed changes appear satisfactory.

BoCui-NOAA commented:
@KateFriedman-NOAA 

Please also update these two scripts:
ush/gfs_sndp.sh
ush/gfs_bfr2gpk.sh

Thanks!

BoCui-NOAA commented:
@KateFriedman-NOAA 

Please switch the owner of the 3 scripts from @ChristopherHill-NOAA to @BoCui-NOAA 
They belong to the bufr sounding package. 
ush/gfs_bfr2gpk.sh
ush/gfs_bufr_netcdf.sh
ush/gfs_bufr.sh

Thanks!

ChristopherHill-NOAA commented:
@KateFriedman-NOAA @WenMeng-NOAA  I agree that @BoCui-NOAA should be listed as the code owner for:
```
ush/gfs_bfr2gpk.sh
ush/gfs_bufr_netcdf.sh
ush/gfs_bufr.sh
```

KateFriedman-NOAA committed:
COM variable updates for bufr and npoess

Refs #2451

KateFriedman-NOAA committed:
Update to CODEOWNERS file for downstream scripts

Refs #2451

KateFriedman-NOAA committed:
Merge remote-tracking branch 'origin/develop' into feature/comin_comout

* origin/develop:
  Create single ocean grib2 product file (#3529)
  Scheduled Nightly CI/CD Pipeline Script in GitLab on Gaea C6 (#3493)
  make sure cold starts are handled correctly when DOIAU=YES (issue #3516) (#3520)
  Add check for DO_AERO_FCST before copying fv_tracer files (#3485)
  Use jinja templates instead of `@VARNAME@` in config files (#3411)
  Replace "status" (or comparable) with "err" in preparation for moving to err_chk/err_exit (#3507)

KateFriedman-NOAA committed:
Remove extra space at end of lines in CODEOWNERS

Refs #2451

KateFriedman-NOAA committed:
Change ATMOS_GRIB from COMOUT to COMIN

Refs #2451

BoCui-NOAA commented on ush/gfs_bfr2gpk.sh:
```diff
@@ -21,14 +21,14 @@ source "${USHgfs}/preamble.sh"
 
 date
-cat "${COM_ATMOS_BUFR}/bufr."*".${PDY}${cyc}" > bufr.combined
+cat "${COMIN_ATMOS_BUFR}/bufr."*".${PDY}${cyc}" > bufr.combined
```
@KateFriedman-NOAA Please change ${COMIN_ATMOS_BUFR} to ${COMOUT_ATMOS_BUFR}

KateFriedman-NOAA replied:
@BoCui-NOAA corrected @ f03d65a2e, thanks!

BoCui-NOAA replied:
Thanks for the updates. The bufr package including the code ownership look good to me.

KateFriedman-NOAA committed:
Change COMIN to COMOUT in gfs_bfr2gpk.sh

Refs #2451

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/comin_comout

KateFriedman-NOAA committed:
Update COM variables in gfs_meta_mar_comp.sh

Refs #2451

KateFriedman-NOAA committed:
Update COM variables in gempak ush scripts

Refs #2451

KateFriedman-NOAA commented:
Rerunning C96_atm3DVar_extended CI test on WCOSS2 to confirm additional updates are good.

KateFriedman-NOAA committed:
Update COMOUT to COMIN in gempak_meta job

Refs #2451

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/comin_comout

KateFriedman-NOAA commented:
Final fixes to gempak_meta job have been committed after successfully completing CI testing on WCOSS2 again. This PR is ready for final reviews and additional CI testing. Thanks all for your comments and feedback thus far!

KateFriedman-NOAA committed:
Merge branch 'develop' into feature/comin_comout

KateFriedman-NOAA committed:
Merge branch 'develop' into feature/comin_comout

DavidHuber-NOAA reviewed (approved):
Changes look good!

DavidHuber-NOAA commented:
Setting the WCOSS2-ready label.

KateFriedman-NOAA commented:
CI Tests set up to run in /lfs/h2/emc/ptmp/emc.global/PR/PR_3518/RUNTESTS on WCOSS

KateFriedman-NOAA commented:
CI completed successfully on WCOSS2:
```
Sun Apr  6 12:58:36 UTC 2025
******** C48_ATM_3518 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Apr 04 2025 19:19:04    Apr 04 2025 20:26:34

******** C48mx500_3DVarAOWCDA_3518 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103241800        Done    Apr 04 2025 19:19:06    Apr 04 2025 19:36:34
202103250000        Done    Apr 04 2025 19:19:06    Apr 04 2025 21:21:36

******** C48mx500_hybAOWCDA_3518 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103241800        Done    Apr 04 2025 19:19:08    Apr 04 2025 19:36:42
202103250000        Done    Apr 04 2025 19:19:08    Apr 04 2025 20:46:37

******** C48_S2SW_3518 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Apr 04 2025 19:19:11    Apr 04 2025 20:36:45

******** C48_S2SWA_gefs_3518 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202103231200        Done    Apr 04 2025 19:19:13    Apr 04 2025 21:06:54

******** C96_atm3DVar_extended_3518 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201800        Done    Apr 04 2025 19:19:16    Apr 04 2025 19:37:07
202112210000        Done    Apr 04 2025 19:19:16    Apr 04 2025 23:46:48
202112210600        Done    Apr 04 2025 19:19:16    Apr 05 2025 00:36:52
202112211200        Done    Apr 04 2025 19:42:03    Apr 05 2025 01:21:59
202112211800        Done    Apr 04 2025 23:51:59    Apr 05 2025 04:42:04

******** C96C48_hybatmaerosnowDA_3518 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201200        Done    Apr 04 2025 19:19:19    Apr 04 2025 19:42:11
202112201800        Done    Apr 04 2025 19:19:19    Apr 04 2025 21:47:01
202112210000        Done    Apr 04 2025 19:19:19    Apr 04 2025 21:37:04

******** C96C48_hybatmDA_3518 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202112201800        Done    Apr 04 2025 19:19:22    Apr 04 2025 19:37:22
202112210000        Done    Apr 04 2025 19:19:22    Apr 04 2025 21:22:16
202112210600        Done    Apr 04 2025 19:19:22    Apr 04 2025 21:27:11

******** C96C48_ufs_hybatmDA_3518 ********
   CYCLE         STATE           ACTIVATED              DEACTIVATED     
202402231800        Done    Apr 04 2025 19:19:24    Apr 04 2025 19:37:29
202402240000        Done    Apr 04 2025 19:19:24    Apr 04 2025 22:17:13
202402240600        Done    Apr 04 2025 19:19:24    Apr 04 2025 22:07:15
```
